src/main/java/uk/gov/ons/ctp/response/action/service/CaseNotificationService.java

```diff
@@ -18,109 +17,143 @@
+      case REPLACED:
+      case ACTIVATED:
+        ActionCase actionCase = createActionCase(notification);
+        saveActionCase(actionCase);
+        break;
+
+      case ACTIONPLAN_CHANGED:
+        updateActionCase(caseId, UUID.fromString(notification.getActionPlanId()));
+        break;
+
+      case DISABLED:
+      case DEACTIVATED:
+        actionService.cancelActions(caseId);
+        deleteActionCase(caseId);
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                INVALID_NOTIFICATION_TYPE, notification.getNotificationType().toString()));
     }
-    final String sampleUnitIdStr = notification.getSampleUnitId();
-    final UUID sampleUnitId = sampleUnitIdStr == null ? null : UUID.fromString(sampleUnitIdStr);
-    final String sampleUnitType = notification.getSampleUnitType();
-
-    if (actionPlan != null) {
-      final ActionCase actionCase =
-          ActionCase.builder()
-              .id(caseId)
-              .sampleUnitId(sampleUnitId)
-              .actionPlanId(actionPlanId)
-              .actionPlanFK(actionPlan.getActionPlanPK())
-              .collectionExerciseId(collectionExerciseId)
-              .partyId(partyId)
-              .sampleUnitType(sampleUnitType)
-              .build();
-
-      switch (notification.getNotificationType()) {
-        case REPLACED:
-        case ACTIVATED:
-          final CollectionExerciseDTO collectionExercise =
-              collectionSvcClientServiceImpl.getCollectionExercise(collectionExerciseId);
-          actionCase.setActionPlanStartDate(
-              new Timestamp(collectionExercise.getScheduledStartDateTime().getTime()));
-          actionCase.setActionPlanEndDate(
-              new Timestamp(collectionExercise.getScheduledEndDateTime().getTime()));
-          checkAndSaveCase(actionCase);
-          break;
-
-        case DISABLED:
-        case DEACTIVATED:
-          try {
-            actionService.cancelActions(caseId);
-          } catch (final CTPException e) {
-            // TODO CTPA-1373 Do we really want to catch this. Should be let to go through.
-            // TODO CTPA-1373 What happens with other notif?
-            log.error("unable to cancel action", e);
-          }
-          final ActionCase actionCaseToDelete = actionCaseRepo.findById(caseId);
-          if (actionCaseToDelete != null) {
-            actionCaseRepo.delete(actionCaseToDelete);
-          } else {
-            log.with("case_id", caseId)
-                .warn("Unexpected situation where actionCaseToDelete is null");
-          }
-          break;
-        default:
-          log.with("notification_type", notification.getNotificationType())
-              .warn("Unknown Case lifecycle event");
-          break;
-      }
-    } else {
-      log.with("action_plan_id", actionPlanIdStr)
-          .warn("Cannot accept CaseNotification for non-existent actionplan");
+    actionCaseRepo.flush();
```

Comment:
In the other methods we have `...Repo.save()` or `...Repo.delete()` called before this
This `flush` is just commiting those changes? or should that happen automatically since this method has the `Transactional` decorator?

Reply:
The changes will be automatically flushed to the DB before the transaction is committed, but I guess there's no harm having this here. The convention everywhere else I've seen has been to flush the changes immediately